fireant/database/base.py

```diff
@@ -66,16 +66,26 @@ def date_add(self, field: terms.Term, date_part: str, interval: int):
         """
         raise NotImplementedError
 
+    def convert_date(self, dt: datetime) -> Union[datetime, Function]:
+        """
+        Override to provide a custom function for converting a date.
+        Defaults to an identity function.
+
+        :param dt: Date to convert
+        """
+        return dt
```

Comment:
MSSQL is a but finicky with datetimes depending on the data type of the underlying datetime column (e.g. datetime vs datetime2), especially when specifying dates as strings (due to different levels of precision). Due to this, I added a convert_date function which defaults to an identity function for all platforms, but is overridden for MSSQL (it converts the date to a `CAST(date as DATETIMEOFFSET)`. 

This is not used inside of fireant, but calling code can convert dates e.g. those that are passed to a BETWEEN query using it. This code doesn't necessarily need to be inside of fireant and can be extracted. The reason I grouped it here is due to the convenience of the database class. Happy to extract it out if people would prefer!